src/TSID/src/QPFixedBaseTSID.cpp

```diff
@@ -0,0 +1,601 @@
+ */
+
+#include <BipedalLocomotion/TSID/QPFixedBaseTSID.h>
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+#include <OsqpEigen/OsqpEigen.h>
+
+#include <BipedalLocomotion/TSID/SE3Task.h>
+#include <BipedalLocomotion/TSID/JointDynamicsTask.h>
+
+#include <BipedalLocomotion/ParametersHandler/StdImplementation.h>
+
+using namespace BipedalLocomotion;
+using namespace BipedalLocomotion::TSID;
+using namespace BipedalLocomotion::ParametersHandler;
+
+struct QPFixedBaseTSID::Impl
+{
+    struct TaskWithPriority
+    {
+        std::shared_ptr<System::LinearTask> task;
+        std::size_t priority;
+        Eigen::VectorXd weight;
+        Eigen::MatrixXd tmp; /**< This is a temporary matrix usefull to reduce dynamics allocation
+                                in advance method */
+    };
+    
+    std::unordered_map<std::string, TaskWithPriority> tasks;
+    std::vector<std::reference_wrapper<const TaskWithPriority>> constraints;
+    std::vector<std::reference_wrapper<TaskWithPriority>> costs;
+    std::size_t numberOfConstraints;
+
+    bool isVerbose{false};
+
+    bool isFirstIteration{true};
+    bool isValid{false};
+    bool isInitialized{false};
+    bool isFinalized{false};
+    bool isKinDynSet{false};
+
+    TSIDState solution;
+
+    System::VariablesHandler::VariableDescription robotAccelerationVariable;
+    System::VariablesHandler::VariableDescription jointTorquesVariable;
+
+    std::string baseLink;
+
+    std::shared_ptr<SE3Task> baseSE3Task;
+    std::shared_ptr<JointDynamicsTask> dynamicsTask;
+
+    OsqpEigen::Solver solver; /**< Optimization solver. */
+
+    Eigen::MatrixXd hessian;
+    Eigen::VectorXd gradient;
+    Eigen::MatrixXd constraintMatrix;
+    Eigen::VectorXd lowerBound;
+    Eigen::VectorXd upperBound;
+
+
+    bool initializeSolver()
+    {
+        constexpr auto logPrefix = "[QPFixedBaseTSID::Impl::initializeSolver]";
+        // Hessian matrix
+        Eigen::SparseMatrix<double> hessianSparse = this->hessian.sparseView();
+        if (!this->solver.data()->setHessianMatrix(hessianSparse))
+        {
+            log()->error("{} Unable to set the hessian matrix.", logPrefix);
+            return false;
+        }
+
+        // gradient
+        if (!this->solver.data()->setGradient(gradient))
+        {
+            log()->error("{} Unable to set the gradient vector.", logPrefix);
+            return false;
+        }
+
+        Eigen::SparseMatrix<double> constraintsMatrixSparse = this->constraintMatrix.sparseView();
+        if (!this->solver.data()->setLinearConstraintsMatrix(constraintsMatrixSparse))
+        {
+            log()->error("{} Unable to set the constraint matrix.", logPrefix);
+            return false;
+        }
+
+        if (!this->solver.data()->setBounds(this->lowerBound, this->upperBound))
+        {
+            log()->error("{} Unable to set the bounds.", logPrefix);
+            return false;
+        }
```

Comment:
can this be replaced by `updateSolver()`?

Reply:
They are not the same. Indeed, here we call `setHessianMatrix`, `setGradient` and so on, instead `updateSolver()` uses `updateHessianMatrix`, `updateGradient`.

Reply:
Ah great, my bad then!